File: galaxy/operator/models/configfile.go
package models

import "time"

type ConfigFileModel struct {
	Pk          string    `json:"pk" db:"pk"`
	Name        string    `json:"name" db:"name"`
	Content     string    `json:"content" db:"content"`
	CreateTime  time.Time `json:"create_time" db:"create_time"`
	UpdateTime  time.Time `json:"update_time" db:"update_time"`
	Description string    `json:"description" db:"description"`
	Mime        string    `json:"mime" db:"mime"`
	Version     int       `json:"version" db:"version"`
	Creator     int64     `json:"creator" db:"creator"`
	Status      int       `json:"status" db:"status"`
}

File: galaxy/web/src/models/configfile.ts
import dayjs from '~/dayjs'
import {ParsedQuery} from '~/query-string'

export interface ConfigFileModel { 
  pk: string
  name: string
  description: string
  create_time: string
  update_time: string
  creator: number
  status: number
  status_name: string
  creator_name: string
}

export class ConfigFileSearchDto {
  page?: number
  size?: number
  creator?: number[]
  keyword?: string
}

export class ConfigFileSearchModel extends ConfigFileSearchDto {
  begin_time_parsed?: dayjs.Dayjs
  end_time_parsed?: dayjs.Dayjs

  static fromDto (query?: ConfigFileSearchDto) {
    const instance = new ConfigFileSearchModel()
    if (!query) return instance
    instance.page = Number(query.page)
    instance.page = Number(query.size)
    return instance
  }

  // 将模型转换为dto对象，确保有合理的默认值，检测边界条件
  static toDto (query: ConfigFileSearchModel | ParsedQuery) {
    const parsedQuery: ConfigFileSearchDto = new ConfigFileSearchDto()
    parsedQuery.page = query.page && !isNaN(Number(query.page)) ? Number(query.page) : undefined
    parsedQuery.size = query.size && !isNaN(Number(query.size)) ? Number(query.size) : undefined

    return parsedQuery
  }

  isEmpty () {
    return !this.page && !this.size
  }
}
